scene/gui/text_edit.cpp

```diff
@@ -953,10 +950,22 @@ void TextEdit::_notification(int p_what) {
 						}
 					}
 
+					//current line highlighting
 					bool in_selection = (selection.active && line >= selection.from_line && line <= selection.to_line && (line > selection.from_line || j >= selection.from_column) && (line < selection.to_line || j < selection.to_column));
 
+					if (line == cursor.line) {
+						if (j == 0)
+							//first char
+							VisualServer::get_singleton()->canvas_item_add_rect(ci, Rect2(0, ofs_y, (char_ofs + char_margin), get_row_height()), cache.current_line_color);
+						else if (j == str.length() - 1)
+							//last char
+							VisualServer::get_singleton()->canvas_item_add_rect(ci, Rect2(char_ofs + char_margin + char_w, ofs_y, xmargin_end - (char_ofs + char_margin + char_w), get_row_height()), cache.current_line_color);
+
+						if (!in_selection)
+							VisualServer::get_singleton()->canvas_item_add_rect(ci, Rect2(Point2i(char_ofs + char_margin, ofs_y), Size2i(char_w, get_row_height())), cache.current_line_color);
+					}
```

Comment:
This will undo the behavior introduced by #11282 hiding the breakpoint marker as it will be overdrawn by the selection/current line.

Reply:
You are refferring to the draw order, right?
Because this changes nothing except the order and that it does not color the whole line but only the part which is not selected.

Reply:
Uh, are you taking into account the fact that it could split the highlight in two if for example just one word is selected?
Also, note that the highlight should be drawn before text (as a background), so if selection color is non-transparent (also as a background), then this wouldn't be needed.

Reply:
@Zylann yea it is drawing two rectangles, one on the left of the selection and one on the right, if that is needed to make the whole line highlighted.
It's also drawn before (draw oder wise) or underneath (I think you get what I mean) the actual font.

Reply:
@toger5 My bad, I had a non-transparent colour so the breakpoint was hidden behind the selection. Tested master and it behaves the same, shows how often I use breakpoints :P

Reply:
@Paulb23 yea I also realized that they were invisible from time to time.
I can look into that in and put breakpoints on top? (drawn after the highlighting)

Oh are we talking about the breakpoints, or the debugging cursor?

Reply:
I think breakpoints must be on top, as if they were characters.
Breakpoint highlight should be on top of line highlight. Well, choose a sensible order :)